### PR TITLE
Redis mirroring should work when default value is zero, not just greater than zero

### PR DESCRIPTION
**Pull request by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg**
Otherwise it will not read runtime settings. This is fixed by checking if the `runtime_fraction` has been set. Also added a test to verify that a default value of zero with no runtime key does not mirror traffic.

Fixes https://github.com/envoyproxy/envoy/issues/8171

### CODE REVIEW

**Comment by mattklein123** [comment]
Thanks for the contribution, test please!

/wait

**Comment by mattklein123** [comment]
@FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg this fixes https://github.com/envoyproxy/envoy/issues/8171 right?

**Comment by FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg** [comment]
@mattklein123 yes, this should fix the linked issue.

File: source/extensions/filters/network/redis_proxy/router_impl.h [hunk 1]
```diff
@@ -37,6 +37,7 @@ class MirrorPolicyImpl : public MirrorPolicy {
 
 private:
   const std::string runtime_key_;
+  const bool has_runtime_fraction_;
   const envoy::type::FractionalPercent default_value_;
```
[the remark this answers is not in the document]
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply source/extensions/filters/network/redis_proxy/router_impl.h]: The default value of default_value is zero, so I need to do a `config.has_runtime_fraction_()` check somewhere. How would you feel about using the optional and doing this check in the body of the constructor?

```
MirrorPolicyImpl::MirrorPolicyImpl(const envoy::config::filter::network::redis_proxy::v2::
                                       RedisProxy::PrefixRoutes::Route::RequestMirrorPolicy& config,
                                   const ConnPool::InstanceSharedPtr upstream,
                                   Runtime::Loader& runtime)
    : runtime_key_(config.runtime_fraction().runtime_key()),
      exclude_read_commands_(config.exclude_read_commands()), upstream_(upstream),
      runtime_(runtime) {
    if (config.has_runtime_fraction()) {
      default_value_ = absl::optional<envoy::type::FractionalPercent>(config.runtime_fraction().default_value());
    }
  }
```

[the remark this answers is not in the document]
Comment (mattklein123) [reply source/extensions/filters/network/redis_proxy/router_impl.h]: You can do it in the initializer list using the ternary operator (and thus make the member variable const). But yeah that's the idea.

[the remark this answers is not in the document]
Comment (FAYiEKcbD0XFqF2QK2E4viAHg8rMm2VbjYKdjTg) [reply source/extensions/filters/network/redis_proxy/router_impl.h]: done

**Pull request merged**